### ISSUE TITLE
Address sanitizer complains about "Heap buffer overflow" (MVSC)

### ISSUE THREAD

**Issue by VitalZero** [raysan5/rres#17]
Using Visual Studio community, raylib 4.2 and c++. Rres file was created using the web version of the app, no encryption or compression was used.

On my resource manager, on the first call to `rresLoadCentralDirectory` throws and exception if I use address sanitizer.
![imagen](https://user-images.githubusercontent.com/46072303/215274451-fd1445d7-68c4-4936-9afa-8a28bbdad0cc.png)

Without address sanitizer, the program runs ok, without any noticeable issues, resources are loaded correctly and displayed/played correctly.

The exception is thrown at line 1075 on rres.h (on the if block).
![imagen](https://user-images.githubusercontent.com/46072303/215273767-2c2f797d-d800-4bfd-9f81-0599fa9cacda.png)
This is the code used on "ResourceManager.h" to call  `rresLoadCentralDirectory`:

```cpp
template<>
inline void ResourceManager::Add<Texture2D>(const std::string& name, const char* path)
{
	if (textures.find(name) == textures.end())
	{
		std::string temp = "Could'nt add the texture ";
		temp += path;

		rresCentralDir texDir = rresLoadCentralDirectory(textureResource.c_str());
		int id = rresGetResourceId(texDir, path);
		rresResourceChunk chunk = rresLoadResourceChunk(textureResource.c_str(), id);

		int result = UnpackResourceChunk(&chunk);

		if (result == RRES_SUCCESS)
		{
			Image img = LoadImageFromResource(chunk);
			bool added = textures.emplace(name, LoadTextureFromImage(img)).second;

			if (!added)
				throw std::runtime_error(temp.c_str());

			UnloadImage(img);
		}

		rresUnloadResourceChunk(chunk);
		rresUnloadCentralDirectory(texDir);
	}
}
```

Code to load the resource:
`data->res.Add<Texture2D>("animation", "newlogo.png");`

Additional info can be found on the attached text file, this was the output generated by address sanitizer log
Thank you Ray!

[rress_ex.txt](https://github.com/raysan5/rres/files/10527534/rress_ex.txt)

**Comment by BeardedBread** [comment]
Hi, I have also encountered this issue on Linux using Clang address sanitizer.

The culprit should be here:
https://github.com/raysan5/rres/blob/536eb301e05e2fafd631e766baeb92d76e6c8f81/src/rres.h#L1073-L1074

From what I understand, info.baseSize is the size of both the properties count, properties array, and the raw data itself; while the input _data_ variable contains all those above. The issue is in the memcpy, where it tries to copy a size more than just the raw data, using the info.baseSize, while the source pointer has been adjusted to the raw data location, causing it to copy beyond the raw data.

I managed to get rid of the issue by computing the raw size before memcpy, modifying the above code like so:
```c
int rawSize = info.baseSize - sizeof(int) - (chunkData.propCount*sizeof(int));
chunkData.raw = RRES_MALLOC(rawSize);
memcpy(chunkData.raw, ((unsigned char *)data) + sizeof(int) + (chunkData.propCount*sizeof(int)), rawSize);
```
